src/components/ComputeModal.js

```diff
@@ -247,9 +269,11 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
     onSuccess()
   })
 
-  const canUpdateNumberOfWorkers = () => {
-    return !currentRuntimeDetails || currentRuntimeDetails.status === 'Running'
-  }
+  const isRuntimeRunning = () => currentRuntimeDetails?.status === 'Running'
+
+  const canManageSparkConsole = () => !!sparkMode && isRuntimeRunning() && computeConfig.componentGatewayEnabled
+
+  const canUpdateNumberOfWorkers = () => !currentRuntimeDetails || isRuntimeRunning()
```

Comment:
Is there a reason these are functions, or could they be constants?

Reply:
Because we want them to be re-evaluated when there are state changes. Does that make sense?